spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/service/impl/TaskServiceUtilsTests.java

```diff
@@ -362,6 +364,54 @@ public void testUseUserAccessTokenFromTaskPropsDefault() {
 		assertFalse("Use user access token should be false", result);
 	}
 
+	@Test
+	public void testConvertCommandLineArgsToCTRFormat() {
+		validateSingleCTRArgs("app.a.0=foo=bar", "--composed-task-app-arguments.base64_YXBwLmEuMA=foo=bar");
+		validateSingleCTRArgs("app.a.0=foo", "--composed-task-app-arguments.base64_YXBwLmEuMA=foo");
+		validateSingleCTRArgs("app.foo.bar", "--composed-task-app-arguments.app.foo.bar");
+		validateSingleCTRArgs("foo=bar", "foo=bar");
+	}
+
+	@Test
+	public void testConvertCommandLineArgsToCTRFormatWithNull() {
+		IllegalArgumentException thrown = Assertions.assertThrows(IllegalArgumentException.class,
+				() -> {
+					List<String> exceptionTestArgs = new ArrayList<>();
+					exceptionTestArgs.add(null);
+					TaskServiceUtils.convertCommandLineArgsToCTRFormat(exceptionTestArgs);
+				}
+		);
+		assertThat(thrown.getMessage()).isEqualTo("Command line Arguments for ComposedTaskRunner contain a null entry.");
+
+	}
+
+	@Test
+	public void testConvertMultipleCommandLineArgsToCTRFormat() {
+		ArrayList<String> originalList = new ArrayList<>();
```

Comment:
**[NIT]** An oldie but a goodie - "[Item 52 - Refer to objects by their interfaces](https://thefinestartist.com/effective-java/52)" 😜